odoo/modules/module.py

```diff
@@ -311,74 +351,85 @@ def get_module_root(path):
+        when the argument is not set.
+    :return dict: The module manifest as a dictionnary or an empty one
+        when the manifest was not found.
     """
     if not mod_path:
         mod_path = get_module_path(module, downloaded=True)
     manifest_file = module_manifest(mod_path)
-    if manifest_file:
-        # default values for descriptor
-        info = {
-            'application': False,
-            'author': 'Odoo S.A.',
-            'auto_install': False,
-            'category': 'Uncategorized',
-            'depends': [],
-            'description': '',
-            'icon': get_module_icon(module),
-            'installable': True,
-            'post_load': None,
-            'version': '1.0',
-            'web': False,
-            'sequence': 100,
-            'summary': '',
-            'website': '',
-        }
-        info.update(zip(
-            'depends data demo test init_xml update_xml demo_xml'.split(),
-            iter(list, None)))
-
-        f = tools.file_open(manifest_file, mode='rb')
-        try:
-            info.update(ast.literal_eval(pycompat.to_text(f.read())))
-        finally:
-            f.close()
-
-        if not info.get('description'):
-            readme_path = [opj(mod_path, x) for x in README
-                           if os.path.isfile(opj(mod_path, x))]
-            if readme_path:
-                with tools.file_open(readme_path[0]) as fd:
-                    info['description'] = fd.read()
-
-        if not info.get('license'):
-            info['license'] = 'LGPL-3'
-            _logger.warning("Missing `license` key in manifest for '%s', defaulting to LGPL-3", module)
-
-        # auto_install is either `False` (by default) in which case the module
-        # is opt-in, either a list of dependencies in which case the module is
-        # automatically installed if all dependencies are (special case: [] to
-        # always install the module), either `True` to auto-install the module
-        # in case all dependencies declared in `depends` are installed.
-        if isinstance(info['auto_install'], collections.abc.Iterable):
-            info['auto_install'] = set(info['auto_install'])
-            non_dependencies = info['auto_install'].difference(info['depends'])
-            assert not non_dependencies,\
-                "auto_install triggers must be dependencies, found " \
-                "non-dependencies [%s] for module %s" % (
-                    ', '.join(non_dependencies), module
-                )
-        elif info['auto_install']:
-            info['auto_install'] = set(info['depends'])
-
-        info['version'] = adapt_version(info['version'])
-        return info
-
-    _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
-    return {}
+
+    if not manifest_file:
+        _logger.debug('module %s: no manifest file found %s', module, MANIFEST_NAMES)
+        return {}
+
+    info = copy.deepcopy(_DEFAULT_MANIFEST)
```

Comment:
Shouldn't `info` be `manifest`?

Reply:
And `deepcopy` wouldn't be necessary if `_DEFAULT_MANIFEST` only contained immutable values (`[]` → `()`, `{}` → `frozendict()`).

Reply:
While I agree with the choice of using immuable data structure, this changes the API of the manifest files. Before it was possible to modify the values of the manifest (e.g. calling `.append` on any list) without changing other manifest files. Using immuable structures (what should had been done in the first place) we break that API.

I decided to be conservative and to use deepcopy() to ensure each default list is independent from one another.

I would prefer to replace those default keys by immutable structure, actually I was secretly hoping one would rise the issue in the review :D

Regarding the point about not-breaking-the-api, do we change it to immutable data structures ?

Reply:
Let's keep the API backward-compatible.
`deepcopy` is actually very reasonable in this situation.